javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+                "        If a message is successfully published to an SNS FIFO topic, any message published and determined to have the same deduplication ID,\n" +
+                "        within the five-minute deduplication interval, is accepted but not delivered.\n" +
+                "        For more information about deduplication, see https://docs.aws.amazon.com/sns/latest/dg/fifo-message-dedup.html.");
+
+            System.out.println("Would you like to use content-based deduplication instead of entering a deduplication ID?");
+            duplication = in.nextLine();
+        }
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("2. Create a topic.");
+        System.out.println("Enter a name for your SNS topic.");
+        topicName = in.nextLine();
+        if (selectFIFO) {
+            System.out.println("Because you have selected a FIFO topic, '.fifo' must be appended to the topic name.");
+            topicName = topicName+".fifo";
+            System.out.println("The name of the topic is "+topicName);
+            topicArn = createFIFO(snsClient, topicName);
+            System.out.println("The ARN of the FIFO topic is "+topicArn);
+
+        } else {
+            System.out.println("The name of the topic is "+topicName);
+            topicArn = createSNSTopic(snsClient, topicName);
+            System.out.println("The ARN of the non-FIFO topic is "+topicArn);
+
+        }
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("3. Create an SQS queue.");
+        System.out.println("Enter a name for your SQS queue.");
+        sqsQueueName = in.nextLine();
+        if (selectFIFO) {
+            sqsQueueName = sqsQueueName + ".fifo";
+        }
+        sqsQueueUrl = createQueue(sqsClient, sqsQueueName, selectFIFO);
+        System.out.println("The queue URL is "+sqsQueueUrl);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("4. Get the SQS queue ARN attribute.");
+        sqsQueueArn = getSQSQueueAttrs(sqsClient, sqsQueueUrl);
+        System.out.println("The ARN of the new queue is "+sqsQueueArn);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("5. Attach an IAM policy to the queue.");
+
+        // Define the policy to use.
+        String policy = "{\n" +
+            "     \"Statement\": [\n" +
+            "     {\n" +
+            "         \"Effect\": \"Allow\",\n" +
+            "                 \"Principal\": {\n" +
+            "             \"Service\": \"sns.amazonaws.com\"\n" +
+            "         },\n" +
+            "         \"Action\": \"sqs:SendMessage\",\n" +
+            "                 \"Resource\": \"arn:aws:sqs:us-east-1:"+accountId+":"+sqsQueueName+"\",\n" +
```

Comment:
Is there a reason you did not use sqsQueueArn for the Resource and topicARN for aws:sourceArn? I think if this is run in a different region than east-1, the code will fail.